File: Easy/0118_pascals-triangle/pascals-triangle.cpp
#include <vector>

class Solution
{
public:
	std::vector<std::vector<int>> generate(int numRows)
	{
		std::vector<std::vector<int>> triangle(numRows);

		triangle[0] = {1};
		for (int i = 1; i < numRows; ++i)
		{
			std::vector<int> row(i + 1);
			row[0] = 1, row[i] = 1;
			for (int j = 1; j < i; ++j)
				row[j] = triangle[i - 1][j] + triangle[i - 1][j - 1];
			triangle[i] = row;
		}
		return triangle;
	}
};

File: Easy/0118_pascals-triangle/pascals-triangle.py
#!/bin/python3

from typing import List

class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        triangle: List[List[int]] = [[1]]

        for _ in range(1, numRows):
            prev = triangle[-1]
            row = [1] + [prev[i] + prev[i+1] for i in range(len(prev)-1)] + [1]
            triangle.append(row)
        return triangle


def main():
    tests: List[int]=[
        5,
        1
    ]

    for rows in tests:
        print(f"---\nnumRows: {rows}\nTriangle: {Solution().generate(rows)}")

if __name__ == "__main__":
    main()
